### PR TITLE
Allow users to update volume name

### PR DESCRIPTION
**Pull request by ravening**
### Description

Provide an api support to update volume name by all users
Can be done both through api and UI

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [X] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [X] Trivial

### Screenshots (if appropriate):

Screenshot 2021-01-14 at 10 25 29

### How Has This Been Tested?

Through API

```
update volume id=3e8f9553-e1c9-4ac0-bb43-10d7d928daf2 name=test-admin
```

### CODE REVIEW

**Review by RodrigoDLopez — CHANGES_REQUESTED** [review]
I am trying to understand in which scenario it is necessary to give freedom for ordinary users to change names within the environment.
You can explain it to me because I couldn't reach a conclusion on my own.

**Review by DaanHoogland — APPROVED** [review]

**Comment by ravening** [comment]
> I am trying to understand in which scenario it is necessary to give freedom for ordinary users to change names within the environment.
> You can explain it to me because I couldn't reach a conclusion on my own.

@RodrigoDLopez there is no harm in changing the volume name. so anybody can do it

File: api/src/main/java/org/apache/cloudstack/api/command/user/volume/UpdateVolumeCmd.java
Comment (ravening) [reply api/src/main/java/org/apache/cloudstack/api/command/user/volume/UpdateVolumeCmd.java:79]: Other parameters have sensitive info whereas the "name" parameter is not critical.
So regular users can change only name but not other important values

File: api/src/main/java/org/apache/cloudstack/api/command/user/volume/UpdateVolumeCmd.java
Comment (ravening) [reply api/src/main/java/org/apache/cloudstack/api/command/user/volume/UpdateVolumeCmd.java:61]: admins and regular users can run this command. but regular users can only change name

**Review by RodrigoDLopez — APPROVED** [review]
@ravening I know that changing volume names does no harm, but I don't like empowering users even if they are cosmetic powers.
But you have my approval and my +1

**Review by weizhouapache — APPROVED** [review]

**Comment by weizhouapache** [comment]
code lgtm

**Comment by mib1185** [comment]
Hi @ravening 
why did you close this PR before merge? 🤔

**Comment by ravening** [comment]
> Hi @ravening
> why did you close this PR before merge? 🤔

@mib1185 sorry it got closed after some rebase with my branch.. I have opened new pr with ui changes in https://github.com/apache/cloudstack/pull/4618

**Pull request closed without merging**
